New package: EmeraldUtilities v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: EmeraldUtilities
- Repository: https://github.com/silicormosia/EmeraldUtilities.jl
- Created by: @Yujie-W
- Version: v0.1.0
- Commit: 6a2e3a6d5ec693359029aab13867fef481a922ed
- Reviewed by: @Yujie-W
- Reference: https://github.com/silicormosia/EmeraldUtilities.jl/commit/6a2e3a6d5ec693359029aab13867fef481a922ed#commitcomment-170764142
- Description: A suite of utility functions to extend the basic packages
<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd6c085db581252f6258844890b0b08b861f8ecf71c102c70c0ba0bb80f9d4d4b45aa88eeb071239df36e72cb4b974606646cd8bfedd3062e4de2403b6cb77e1334cb9dabf163ac6cf2a62564fb764cc9f9a146b6c95decf055dbb9958833b95d6382cf00884077464496f1b427efc49efa7980be8a8a440b64263f9437c7e23c167b354bfd50dde59e9f6e796ce0783bd3df6c99405bf7caf527f4e57c4cda0af603fd1b242f34372cfb2ba5b486e57dc4f3b6e16ac6a03459f02e0ffaa5c56dc6 -->

goerz commented:
Thank you for submitting your package! Could you add a little bit of documentation before registering? [At the very least](https://discourse.julialang.org/t/when-does-it-make-sense-to-post-a-new-package-to-the-julia-repository/108701/4), that would be a clear description of the package's purpose and a small usage example in the README. An important part of packages in General is that any potential user can figure out what the package is about and how to get started with using it. That is really difficult when there is no documentation.

In the longer term, I definitely recommend setting up a [Documenter](https://documenter.juliadocs.org/stable/)-based documentation. Before a `v1.0` release, or for smaller packages that can be effectively described entirely with their `README`, that's not a requirement, though.

I would also mention that "personal utility packages" without a clear scope are note suitable for registration according to the [guidelines](https://github.com/JuliaRegistries/General?tab=readme-ov-file#should-i-register-my-package-now). Without documentation, it is not clear whether this package would fall in that category. What are "the basic packages"?

Yujie-W commented:
Since the dirst registration requires 3 days to proceed, I wanna start the thread earlier so that I can reuse the package in others earlier. The documentation has been added already for each submodule, but not yet deployed on the website.
E.g., https://github.com/silicormosia/EmeraldUtilities.jl/blob/3bfb404e7a619e4d0e43c45ed70bb42644f1d1c7/src/RecursiveTools/struct-compare.jl#L10

JuliaRegistrator committed:
New package: EmeraldUtilities v0.1.0

UUID: bf152e34-6238-4522-95c8-8e628351fb48
Repo: https://github.com/silicormosia/EmeraldUtilities.jl.git
Tree: 5501c678e54bc2cb169701f8eea3822963bb6027

Registrator tree SHA: 50f504d641745716a5b3eabaf681d3a4937d2ae3

Yujie-W commented:
> In the longer term, I definitely recommend setting up a [Documenter](https://documenter.juliadocs.org/stable/)-based documentation. Before a `v1.0` release, or for smaller packages that can be effectively described entirely with their `README`, that's not a requirement, though.

[noblock]
Detailed documentation and tests added now, and please continue the merging process
